### ISSUE TITLE
[Desktop-Windows]Chinese font looks weird in flutter 3.0.0

### ISSUE THREAD

**Issue by Kamihimmel** [flutter/flutter#103811]
Chinese font flutter 3.0.0 Desktop-Windows
![QQ截图20220514235906destop2](https://user-images.githubusercontent.com/7413008/168432499-ac1a22cd-8abb-44b6-a587-8972c2f1cd17.jpg)

Chinese font on flutter 2.10.5 Desktop-Windows
![QQ截图20220514235727desktop1](https://user-images.githubusercontent.com/7413008/168432594-5e3d1bac-562c-4ccd-a0a9-671e6989fd3f.jpg)

This is a desktop specific issue.(this is good on web)  flutter 3.0.0 web
![QQ截图20220514235811web1](https://user-images.githubusercontent.com/7413008/168432751-f7916f07-ac63-449a-8ef1-dcd0176589b3.jpg)

**Comment by hhgz9527** [comment]
Maybe same issue https://github.com/flutter/flutter/issues/103854

**Comment by maheshmnj** [comment]
Hi @Kamihimmel, Thanks for filing the issue. Although I see slight difference in both the text. I tried to compare the outputs of a Web and Desktop by running the counter app with Chinese text and I could not see the difference.
  
Image 1             |  Image2
:-------------------------:|:-------------------------:
<img width="648" alt="image" src="https://user-images.githubusercontent.com/31410839/168558841-41cf72a8-ee49-4d67-a847-126bd7edd70c.png"> | <img width="677" alt="image" src="https://user-images.githubusercontent.com/31410839/168558712-938217fd-5987-49ff-9d35-44e7c50aeb11.png">

#### Counter app with Chinese text

<img width="1358" alt="image" src="https://user-images.githubusercontent.com/31410839/168569317-76b7d78b-b81a-4ffc-af41-724fc52f846d.png">


Does your app use any Custom font? Can you please share the output of `flutter doctor -v`?

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.0.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by fantasy12o3** [comment]
> Hi @Kamihimmel, Thanks for filing the issue. Although I see slight difference in both the text. I tried to compare the outputs of a Web and Desktop by running the counter app with Chinese text and I could not see the difference.
> 
> Image 1	Image2
> <img alt="image" width="648" src="https://user-images.githubusercontent.com/31410839/168558841-41cf72a8-ee49-4d67-a847-126bd7edd70c.png">	<img alt="image" width="677" src="https://user-images.githubusercontent.com/31410839/168558712-938217fd-5987-49ff-9d35-44e7c50aeb11.png">
> #### Counter app with Chinese text
> <img alt="image" width="1358" src="https://user-images.githubusercontent.com/31410839/168569317-76b7d78b-b81a-4ffc-af41-724fc52f846d.png">
> 
> Does your app use any Custom font? Can you please share the output of `flutter doctor -v`?
> 
> flutter doctor -v (mac)

for bold Chinese text, not all characters render as the same. It seems the font use for simplified Chinese is a font only for Traditional Chinese

**Comment by Kamihimmel** [comment]
@maheshmnj 
`[√] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Version 10.0.19044.1706], locale zh-CN)
    • Flutter version 3.0.0 at C:\src\flutter\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (7 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\Owner\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.11)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.32228.343
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] VS Code (version 1.67.1)
    • VS Code at C:\Users\Owner\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.1706]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.67
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 100.0.1185.44

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!`

No custom font is used in my app. Nor adding any font weight to text widget.
Your demo are only comparing web with MacOS desktop, which is not the same case with Windows desktop. As the typography is different between MacOS and Windows.

**Comment by Kamihimmel** [comment]
@maheshmnj  And not all characters are bugged. It seems like that flutter is having a difficult recognizing traditional and simplified Chinese. You can try "雷电将军" for testing.

**Comment by maheshmnj** [comment]
Thanks for the info @Kamihimmel, To be more specific is this issue with the first character in "雷电将军" displayed slightly differently on the web and Desktop?

  ### Mac
![image](https://user-images.githubusercontent.com/31410839/168737305-470d9723-75bc-418a-8d6f-81ef244e602c.png)

### Windows
![image](https://user-images.githubusercontent.com/31410839/168738937-7ba7b4ec-ac44-48fe-b1ee-22c6b746b228.png)
<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.0.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

<details>
<summary>flutter doctor -v (windows)</summary>

```
[√] Flutter (Channel stable, 3.0.0, on Microsoft Windows [Version
    10.0.19044.1645], locale en-IN)
    • Flutter version 3.0.0 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)  
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual
      Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)    

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.40.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 
      10.0.19044.1645]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 101.0.4951.54    • Edge (web)        • edge    • web-javascript • Microsoft Edge
      100.0.1185.50

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by fantasy12o3** [comment]
![微信图片_20220517151051](https://user-images.githubusercontent.com/8768414/168750990-872aa7af-0fcb-4b95-9e88-a63e917d7bb5.png)

only the bold text .  固件升级. also the font used for normal text is rarely used for  simplified Chinese

**Comment by Kamihimmel** [comment]
@maheshmnj not actually the bug I want to report.. The problem you discover is a different bug btw(it exists before 3.0.0, if you like you can make that a new ticket as well).
(My problem is like @fantasy12o3 reports) Can you use a text widget without any font size or font weight with the same text?
my code is like below 

```
ListTile(
              title: Text('雷电将军'),
            ),
```

**Comment by maheshmnj** [comment]
@Kamihimmel, I do see the difference in font weight for non Chinese (english) characters on web and Desktop but not across different characters. However, the first letter seems inconsistent across web and Desktop as [mentioned here](https://github.com/flutter/flutter/issues/103811#issuecomment-1128461878)

<img width="1433" alt="image" src="https://user-images.githubusercontent.com/31410839/168834609-60fcbb05-d7f8-41ab-9d50-58b17615fa6b.png">

Labeling this issue for further investigation from the team

<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.0, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 3.0.0 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee4e09cce0 (6 days ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```
    • Flutter version 3.1.0-0.0.pre.419 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2eed8cbf93 (2 weeks ago), 2022-04-28 22:29:06 -0400
    • Engine revision dfdfe0b3b0
    • Dart version 2.18.0 (build 2.18.0-66.0.dev)
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! CocoaPods 1.10.2 out of date (1.11.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your
        plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3 21E230 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

**Comment by gspencergoog** [comment]
For the time being, for desktop, you can specify the exact font in your `pubspec.yaml`, to make sure you get the font you want.

cc @dkwingsmt

**Comment by Kamihimmel** [comment]
> For the time being, for desktop, you can specify the exact font in your `pubspec.yaml`, to make sure you get the font you want.
> 
> 
> 
> cc @dkwingsmt 

Thanks for the workaround. However for many non-Latin languages, font itself takes a lot of space. For Chinese font it takes 6-10M for font itself.Which makes it impossible to embed it in web.For supporting multiple languages means more font files. 
My workaround to this problem is change locale from Zh-Hans to Zh-Hans-CN. But in this way other non-cn Zh-Hans locale won't be supported.It would be better if flutter let us choose what system font to use.

**Comment by lastwildor** [comment]
![image](https://user-images.githubusercontent.com/56182468/170815722-9c7371c9-9e8b-4f1f-902d-c861434c4ca1.png)
After I upgraded to flutter 3.0, I also have this problem on windows

**Comment by dkwingsmt** [comment]
我理解这些字体是简繁字体的问题，但我无法复现。我甚至专门去搭了个新Windows 10的虚拟机在里面重新开flutter环境运行代码，也无法复现这个问题。下面是我的代码：
```dart
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        body: Column(
          children: <Widget>[
            ListTile(
              title: Text(
                '雷电将军 升级 你好'
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

flutter doctor -v
<details><pre>
PS C:\Users\dzltm\ftest> flutter doctor -v     
[√] Flutter (Channel unknown, 3.0.0, on Microsoft Windows [Version 10.0.19043.928], locale zh-CN)
    • Flutter version 3.0.0 at C:\Users\dzltm\flutter
    • Upstream repository unknown
    • Framework revision ee4e09cce0 (3 weeks ago), 2022-05-09 16:45:18 -0700
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)    
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.2)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32519.379
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] VS Code (version 1.67.2)
    • VS Code at C:\Users\dzltm\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension can be installed from:
       https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[√] Connected device (2 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.928]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 101.0.1210.53

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 3 categories.
</pre></details>

我在思考可能和windows语言有关，能不能截一下下面这个图？控制面板-时间和语言-语言。
![image](https://user-images.githubusercontent.com/1596656/170821937-562dc229-232f-4e8c-8497-f4db111ff369.png)

还有下面这个，在右边的管理语言设置里：
![image](https://user-images.githubusercontent.com/1596656/170821968-a5e26be8-45e5-4d05-b542-13146172253a.png)

**Comment by lastwildor** [comment]
我实验了下，直接用Text()显示的是下面的效果

![image](https://user-images.githubusercontent.com/56182468/170823328-09fa20bb-2474-42e5-bf6d-6cd1dcea8f01.png)

如果包裹了TextButton，child下的Text就是就会乱变加粗效果了。如下图：
![image](https://user-images.githubusercontent.com/56182468/170823357-8fab6fe9-eeec-44ea-a2cf-61292c52dffd.png)

如果改了下全局textButtonTheme的皮肤好像要好些了。
============

但总体来说，flutter3下的windows中文字体简直惨不忍睹。，根本不敢去上市面对用户啊。不行只有回退老版本了。我不用flutter自带字体，自己直接用本地黑体字体。效果还是不理想，感觉锯齿感很强！

**Comment by dkwingsmt** [comment]
感谢，我这里能复现了，我去查一下是哪里出的问题。

**Comment by dkwingsmt** [comment]
With bisection, this error is caused by https://github.com/flutter/flutter/pull/97180, an engine roll that contains only one engine PR, https://github.com/flutter/engine/pull/28912.

@jason-simmons

**Comment by dkwingsmt** [comment]
这个选项是可以关闭的，但这个关闭的选项没有直接暴露出来。暂时可以用以下解决方案：

1. 编辑[desktop_device.dart#L234](https://github.com/flutter/flutter/blob/afca4fb248f2814624b2503f7805e9b8477f4678/packages/flutter_tools/lib/src/desktop_device.dart#L234)
```diff
    addFlag('enable-dart-profiling=true');
+   addFlag('enable-skparagraph=false');
```

2. 根据[flutter_tools/README.md](https://github.com/flutter/flutter/tree/afca4fb248f2814624b2503f7805e9b8477f4678/packages/flutter_tools)的指引，使用`dart bin/flutter_tools.dart`替换`flutter`命令行指令。

**Comment by Kamihimmel** [comment]
> 我理解这些字体是简繁字体的问题，但我无法复现。我甚至专门去搭了个新Windows 10的虚拟机在里面重新开flutter环境运行代码，也无法复现这个问题。下面是我的代码：
> 
> ```dart
> 
> class MyApp extends StatelessWidget {
> 
>   @override
> 
>   Widget build(BuildContext context) {
> 
>     return MaterialApp(
> 
>       title: 'Flutter Demo',
> 
>       theme: ThemeData(
> 
>         primarySwatch: Colors.blue,
> 
>       ),
> 
>       home: Scaffold(
> 
>         body: Column(
> 
>           children: <Widget>[
> 
>             ListTile(
> 
>               title: Text(
> 
>                 '雷电将军 升级 你好'
> 
>               ),
> 
>             ),
> 
>             ListTile(
> 
>               title: Text(
> 
>                 '雷电将军 升级 你好',
> 
>                 style: TextStyle(fontWeight: FontWeight.bold),
> 
>               ),
> 
>             ),
> 
>           ],
> 
>         ),
> 
>       ),
> 
>     );
> 
>   }
> 
> }
> 
> ```
> 
> 
> 
> 出来的图是这样，虽然确实有粗细不一致的情况……但应该没有到最开始那张图的程度。
> 
> ![image](https://user-images.githubusercontent.com/1596656/170822040-5f983fdc-ddc4-4b0a-8b58-1fa3ccc2b01e.png)
> 
> 
> 
> 
> 
> flutter doctor -v
> 
> <details><pre>
> 
> PS C:\Users\dzltm\ftest> flutter doctor -v     
> 
> [√] Flutter (Channel unknown, 3.0.0, on Microsoft Windows [Version 10.0.19043.928], locale zh-CN)
> 
>     • Flutter version 3.0.0 at C:\Users\dzltm\flutter
> 
>     • Upstream repository unknown
> 
>     • Framework revision ee4e09cce0 (3 weeks ago), 2022-05-09 16:45:18 -0700
> 
>     • Engine revision d1b9a6938a
> 
>     • Dart version 2.17.0
> 
>     • DevTools version 2.12.2
> 
> 
> 
> [X] Android toolchain - develop for Android devices
> 
>     X Unable to locate Android SDK.
> 
>       Install Android Studio from: https://developer.android.com/studio/index.html
> 
>       On first launch it will assist you in installing the Android SDK components.
> 
>       (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
> 
>       If the Android SDK has been installed to a custom location, please use
> 
>       `flutter config --android-sdk` to update to that location.
> 
> 
> 
> 
> 
> [X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)    
> 
>     ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
> 
> 
> 
> [√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.2)
> 
>     • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
> 
>     • Visual Studio Community 2022 version 17.2.32519.379
> 
>     • Windows 10 SDK version 10.0.19041.0
> 
> 
> 
> [!] Android Studio (not installed)
> 
>     • Android Studio not found; download from https://developer.android.com/studio/index.html
> 
>       (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
> 
> 
> 
> [√] VS Code (version 1.67.2)
> 
>     • VS Code at C:\Users\dzltm\AppData\Local\Programs\Microsoft VS Code
> 
>     • Flutter extension can be installed from:
> 
>        https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter
> 
> 
> 
> [√] Connected device (2 available)
> 
>     • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.928]
> 
>     • Edge (web)        • edge    • web-javascript • Microsoft Edge 101.0.1210.53
> 
> 
> 
> [√] HTTP Host Availability
> 
>     • All required HTTP hosts are available
> 
> 
> 
> ! Doctor found issues in 3 categories.
> 
> </pre></details>
> 
> 
> 
> 我在思考可能和windows语言有关，能不能截一下下面这个图？控制面板-时间和语言-语言。
> 
> ![image](https://user-images.githubusercontent.com/1596656/170821937-562dc229-232f-4e8c-8497-f4db111ff369.png)
> 
> 
> 
> 还有下面这个，在右边的管理语言设置里：
> 
> ![image](https://user-images.githubusercontent.com/1596656/170821968-a5e26be8-45e5-4d05-b542-13146172253a.png)
> 
> 
> 
> 

我想问一下 这个将 和你回复的将写法完全不同。是因为flutter3.0变的默认字体变了么？由默认简体变成了默认繁体？这个和上面的粗体感觉不是一个问题。

**Comment by dkwingsmt** [comment]
原因大概是使用了不同的字体渲染方式（SkParagraph），至于为什么这会导致这些问题超出了我的知识范围，可能需要jason来调查。

**Comment by litang0908** [comment]
Same problem , so my solution is replace the font with another one :

fonts in pubspec.yaml
```
fonts:
    - family: MiSans
      fonts:
        - asset: assets/fonts/MiSans-Regular.ttf
        - asset: assets/fonts/MiSans-Demibold.ttf
```

**Comment by dkwingsmt** [comment]
@Kamihimmel I just realized what you're saying with the different shape of "将". However I can't even print the correct shape under 2.10.5. Can you provide a minimal example for that? 

For reference, this is the app I used:
```dart
import 'package:flutter/material.dart';

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Column(
            children: <Widget>[
              TextButton(
                child: Text('雷电将军 升级 你好 道场'),
                onPressed: () {},
              ),
              Text('雷电将军 升级 你好 道场'),
            ],
          )
          ,
        ),
      ),
    );
  }
}
```

And this is what I get on 2.10.5:

![image](https://user-images.githubusercontent.com/1596656/171272578-5b7bfaff-d0ea-40e7-a784-86976a508266.png)

**Comment by jason-simmons** [comment]
This is happening because the `TextButton` example is using a style that specifies `FontWeight.w500`, but the Windows host does not provide Chinese fonts with that exact weight.  

SkParagraph needs to do font fallback and query the system for the closest available fonts.  However, the queries are selecting different fonts with inconsistent weights.

On my Windows setup, the first character in the string is handled by the "Yu Gothic UI" font.  That font is available in weights 300, 400, 600, and 700.  The Skia `SkFontMgr` implementation selects the font with weight 600 (both 400 and 600 are equally close to the desired weight 500).

The next character is not handled by that font, so SkParagraph needs to find a different font.  It finds "Microsoft JhengHei UI", which is available in weights 200, 400, and 700.  The `SkFontMgr` selects the version with weight 400 (which is closest to 500).

Therefore you see a mix of weight 600 and weight 400 glyphs, resulting in the effect seen in the screenshots above.

The selection of the closest font for each character is done by `SkFontMgr_DirectWrite::fallback`, which calls the Windows `IDWriteFontFallback::MapCharacters` API.  If two fonts are available that are equally close matches to the specified weight, then apparently that API prefers the font with the higher weight.

This is different from the CSS3 font matching algorithm which is used by Skia font managers on some other Flutter platforms.  https://www.w3.org/TR/css-fonts-3/#font-style-matching specifies that "If the desired weight is 500, 400 is checked first and then the rule for desired weights less than 400 is used."

See https://bugs.chromium.org/p/chromium/issues/detail?id=1022782#c14 for another example of this.

To work around this, the app could specify styles that use `FontWeight.w400` instead of `w500`.

Or it could disable SkParagraph and revert to the old Libtxt text renderer.

**Comment by Kamihimmel** [comment]
@jason-simmons @dkwingsmt Sorry to mention you two together.But the explanation above explains the issue well. 

> On my Windows setup, the first character in the string is handled by the "Yu Gothic UI" font. That font is available in weights 300, 400, 600, and 700. The Skia SkFontMgr implementation selects the font with weight 600 (both 400 and 600 are equally close to the desired weight 500).

> The next character is not handled by that font, so SkParagraph needs to find a different font. It finds "Microsoft JhengHei UI", which is available in weights 200, 400, and 700. The SkFontMgr selects the version with weight 400 (which is closest to 500).

Now my question is, there is 2 fonts being used in one sentences.And "Yu Gothic UI" is a Japanese font and "Microsoft JhengHei UI" is a Chinese font.I don't know a lot about how fonts are matched. But the intent of author (me) is this is a Chinese sentence. Chinese font will have all of the characters and Japanese font wont have all of them.And the system somehow prioritize Japanese over Chinese. Is that true?
Which is an absurd behavior, as on the web(now in github), these issue will not happen.
@dkwingsmt You can set the local to Zh-hans to retest it and for after 3.0 use Zh-hans-cn.

My pc locale is China, and I have to forcefully set locale to "Zh-hans-cn" to make all the characters right.
These problem occurs since day one and may be a different problem.

**Comment by w568w** [comment]
Any progress on the issue? So far I have to roll back to 2.x version to release applications for Windows.

As @Kamihimmel said, there are **two** different problems since 3.0, which are:
1. To satisfy the font weight `w500` requirement, SkParagraph is selecting different fonts for a single sentence.
2. SkParagraph prefers Japanese fonts to others. In fact, it seems that the render applies JP fonts to *any* CJK characters, even under an environment with `zh-cn` (Simplified Chinese) locale. 

Chinese fonts are [visually different from their Japanese counterparts](https://blog.skritter.com/2015/06/font-differences-between-japanese-and-chinese/), thus they cannot replace each other.

**Comment by cbracken** [comment]
@jason-simmons what are the next steps required here? If the behaviour of SkParagraph is inconsistent with libtxt, it would be nice to either correct that in SkParagraph or something on our side to match the old behaviour.

> Flutter 3.0+ prefers Japanese fonts to others. 

As far as I know we have no preferred locale ordering in Flutter itself, but rather just use the system font fallback ordering. 

Out of curiosity, what happens when you set [`Text.locale`](https://api.flutter.dev/flutter/widgets/Text/locale.html) on the text in question?

**Comment by jason-simmons** [comment]
We need to work with Skia to find a solution for this.

SkParagraph uses Skia's `SkFontMgr::matchFamilyStyleCharacter` API to find a font that can handle a character that is not supported by the `TextStyle`'s specified font families.

Each platform's `matchFamilyStyleCharacter` implementation tries to find a font that includes the requested character and also matches the `SkFontStyle` attributes as closely as possible.  

On some platforms `matchFamilyStyleCharacter` may not find the closest match for the style attributes (see https://github.com/flutter/flutter/issues/105014).  Or it may choose a match that differs from the one that had been selected by Libtxt in older versions of Flutter (as in this issue).

Libtxt had been using `matchFamilyStyleCharacter` to find a font that supports the given character.  It then used CSS3-style font matching to select the closest match for the style attributes within that font's family

That policy generally worked well, and I believe it is similar to what Chromium does.  However, I think Skia was concerned that this could be inaccurate because other fonts with the same family name do not necessarily cover the same characters.

SkParagraph may need to provide hooks that let users like Flutter customize the font selection policy.

Another option that might work would be wrapping the platform's `SkFontMgr` with something that adds the desired policy.  I've also been considering that as a way to force selection of a particular emoji font on some platforms to work around issues like https://github.com/flutter/flutter/issues/100964.

**Comment by dkwingsmt** [comment]
@jason-simmons Before that is possible, can we have a flag to switch to the legacy behavior to unblock publishing Chinese apps?

**Comment by w568w** [comment]
> Out of curiosity, what happens when you set [`Text.locale`](https://api.flutter.dev/flutter/widgets/Text/locale.html) on the text in question?

The widgets seem to choose the right font after setting `Text.locale`. 

Code:
```dart
Text(
  "With Locale:\nLorem Ipsum Lorem Ipsum\n共技刻只言 共技刻只言",
  locale: Locale("zh", "CN"),
),
Text(
  "Without Locale:\nLorem Ipsum Lorem Ipsum\n共技刻只言 共技刻只言",
),
```
Screenshot:

![image](https://user-images.githubusercontent.com/8389787/181711716-ca413473-78e7-4cae-84b9-9755c2f5ad92.png)

Sorry if I've made a mistake in the description. "Under an environment with `zh-cn` (Simplified Chinese) locale" in my last comment refers to that in [the system locale setting](https://support.microsoft.com/help/4496404), not in Flutter.

# A workaround
The attempt inspired me. As far as I can see, [`Text.locale`](https://api.flutter.dev/flutter/widgets/Text/locale.html) is said to be "inherited from the enclosing app with `Localizations.localeOf(context)`", which produces `en-US` in most cases, since its `localizationsDelegates` do not support `zh-*` locale by default. **So Flutter always falls back on `en-US` and selects different fonts.**

A quick workaround is to [enable internationalization](https://docs.flutter.dev/development/accessibility-and-localization/internationalization). After following the documentation, declare `supportedLocales` in the root `WidgetApp` so that Flutter will not use the default locale, i.e.
```dart
return MaterialApp(
  title: 'Demo',
  localizationsDelegates: const [
    GlobalMaterialLocalizations.delegate,
    GlobalWidgetsLocalizations.delegate,
    GlobalCupertinoLocalizations.delegate,
  ],
  // Insert this line
  supportedLocales: const [Locale("zh", "CN"), Locale("en", "US")],
  home: const MyHomePage(),
);
```
Then everything looks fine:

![image](https://user-images.githubusercontent.com/8389787/181716868-b76f587c-d5fd-4bc8-abaa-c18890f67845.png)

**Comment by Ronbb** [comment]
You can set default locale from system.

```dart
MaterialApp(
  // ...
  // got 'zh'
  locale: Locale(Platform.localeName.split('_')[0]),
  // ...
);
```

It works with Flutter 3.0.5.

**Comment by dkwingsmt** [comment]
Thanks. Since there's a workaround, I guess this no longer counts as a blocker.

**Comment by SNNafi** [comment]
@dkwingsmt, I hope you are well. I have edited the `desktop_device.dart` as you mentioned earlier and run `dart bin/flutter_tools.dart`

Then, rebuild project, run in both macOS and windows, still no luck. 

However, Worked in android using,
```
<meta-data
    android:name="io.flutter.embedding.android.EnableSkParagraph"
    android:value="false" />
    
``` 

What can I do now?

**Comment by yaakovschectman** [comment]
When a locale is not specified for a `Text`, the first locale that is present in both the preferred and supported locales is used for the widget. By default, this will be `en-US`, which renders characters in one way; other locales like `zh-CN` won't be used unless explicitly specified for the widget or `supportedLocales` is provided and does not include `en-US`.

**Comment by YI--** [comment]
Are there any new solutions or official updates regarding this issue, aside from https://github.com/flutter/flutter/issues/103811#issuecomment-1199012026?

**Comment by SekiBetu** [comment]
any updates?

**Comment by MegatronKing** [comment]
I use flutter v3.10, the height of the Chinese font cannot be aligned. There are some slight offsets, need to look carefully, and only happens on Windows. Also, when on a high-resolution screen, adjusting system scaling can be very noticeable.

<img width="293" alt="截屏2023-09-06 23 56 36" src="https://github.com/flutter/flutter/assets/15059084/a4ee0330-d101-415f-9778-6954a3c3ad52">

**Comment by lucasjinreal** [comment]
我也遇到了通用的问题，默认的Flutter windows，简体中文的字体看起来非常的不正常，看起来是一个繁体字体，显示粗糙不锐利，看起来也不清晰，到目前该问题还没有解决？

![image](https://github.com/flutter/flutter/assets/21303438/c8a39b64-66f7-4cd3-8d9d-9aa048c84633)

 I also encountered a general problem, the default Flutter windows, the simplified Chinese font looks very abnormal, it looks like a traditional Chinese font, and the display is rough and not sharp, it does not look clear, and so far, this problem has not been solved?

**Comment by lucasjinreal** [comment]
I set the locale Zh first, it now looks a little bit normal, but still not very clear, weired.

**Comment by ChenDoXiu** [comment]
After I manually specified the Text font as Microsoft Yahei, the problem was temporarily solved.
```dart
fontFamily: Platform.isWindows ? "微软雅黑" : null,
```

**Comment by cbracken** [comment]
One more ping @Rusino for thoughts from the SkParagrap font fallback side. https://github.com/flutter/flutter/issues/103811#issuecomment-1198547719 discusses the font fallback difference we're seeing since moving to SkParagraph from libtxt. Thoughts?

**Comment by Rusino** [comment]
This issue is platform dependent. Method "matchFamilyStyleCharacter" mentioned above depends on a platform (CoreText for Apple, DirectWrite for Windows,...). Whatever the platform provides we show. It is absolutely impossible for us to show something else - just because it would mean to substitute the platform functionality (loading and parsing all the fonts ahead of time and such). 
Even the difference between libtxt and SkParagraph: the newer version of CoreText gives different match results and the last you have seen on libtxt worked on the old version.
What I am trying to say is that we cannot fix it. Anything we do here would be finding another heuristic that would cover some of the cases but not all of them.
Does Flutter allow a user to set a default fallback font? It would help with this issue.

**Comment by cbracken** [comment]
Thanks @Rusino that's really useful context. What we've been asking users to do is [set the region/script explicitly](https://github.com/flutter/flutter/issues/103811#issuecomment-1199012026), which to be honest is probably the only 100% guaranteed solution since it's quite common for many people to run apps in one language on a device set to another language/script.

In the vein of your suggestion, it sounds like the followup here is for us to investigate what means we can provide on the Flutter side to make setting this info less of a pain to the developer.

**Comment by rockingdice** [comment]
> This is because of the font, “Segoe UI” doesn't support Chinese and flutter does not fallback to "Microsoft YaHei", flutter uses “Segoe UI” in windows by default. In web platform, it fallback to "Microsoft YaHei" in windows. try this code:
> 
> Column(
>   crossAxisAlignment: CrossAxisAlignment.start,
>   children: [
>     Text("启动(fontFamily: Microsoft YaHei)", style: TextStyle(fontFamily: "Microsoft YaHei")),
>     Text("启动(fontFamily: Segoe UI)", style: TextStyle(fontFamily: "Segoe UI")),
>     Text("启动(fontFamily: Default)"),
>   ],
> ),
> ## windows desktop
> ![Image](https://github.com/user-attachments/assets/cbf3c06a-8090-4520-8a9a-bd6a22e79a17)
> 
> # windows web
> ![Image](https://github.com/user-attachments/assets/1e125676-9b49-4ab1-aa77-a4555692211d)
> 
> Windows settings indicate that “Segoe UI” does not support Chinese.
> 
> ![Image](https://github.com/user-attachments/assets/5d13be5c-19ef-4ae0-96ba-658140db0aec)

A life saver! thank you for the info, it's a big improvement for changing the default fonts, thank you!
Though Can you make the fallback font to Microsoft Yahei on the windows platform?

**Comment by brandon3343** [comment]
Yes, you can achieve this effect by replacing the font settings in the theme.
For example, the following code will display correctly with no display issues.
```dart
Text("启动", style: TextStyle(fontFamilyFallback: ["Microsoft YaHei"]) )
```

**Comment by w568w** [comment]
@brandon3343 It looks like older comments are collapsed, so later readers might have missed the solution suggested earlier. 

Have you tried https://github.com/flutter/flutter/issues/103811#issuecomment-1199012026? Does it work better or worse?

**Comment by brandon3343** [comment]
No, I use custom fonts in my app to get visual consistency across all platforms.

**Comment by Zhoucheng133** [comment]
Here's the simplest solution:

You can use `google_fonts` dependency and set it to default font.

1. `flutter pub add google_fonts`
2. Set it to default font:
```dart
return MaterialApp(
  theme: ThemeData(
    textTheme: GoogleFonts.notoSansScTextTheme(),
  ),
);
```
Then you need NOT to set every `Text` font to Noto Sans

OR if the UI framework (Such as Fluent UI) does not support set default `textTheme`, you can set every Text as follows:
```dart
return Text(
  "测试文本",
  style: GoogleFonts.notoSansSc(),
);
```

By the way, even if google_fonts will download the font online, it works in mainland China as well

**Comment by wangrui112211** [comment]
![Image](https://github.com/user-attachments/assets/d3b0a6ae-1ef1-43be-80c2-30bf859aff83)flutter3.38.3在小米手机发现的问题，其他机型没问题,中文的中字跟其他字显示的不一样

**Comment by wangrui112211** [comment]
> ![Image](https://github.com/user-attachments/assets/d3b0a6ae-1ef1-43be-80c2-30bf859aff83)flutter3.38.3在小米手机发现的问题，其他机型没问题,中文的中字跟其他字显示的不一样



![Image](https://github.com/user-attachments/assets/76c19fc0-e24c-499e-b717-053776e5372e)

三星手机的testSpan同样式的字体显示也不一样
